cluster/clustered_service_agent.go

```diff
@@ -774,10 +795,19 @@ func (agent *ClusteredServiceAgent) offerToSession(
 	length int32,
 	reservedValueSupplier term.ReservedValueSupplier,
 ) int64 {
+	if err := agent.checkForValidInvocation(); err != nil {
+		logger.Errorf("offerToSession: error in checkForValidInvocation clusterSessionId=%d publication.sessionID=%d offset=%d length=%d", clusterSessionId, publication.SessionID(), offset, length)
+		return NullValue
+	}
+
 	if agent.role != Leader {
 		return ClientSessionMockedOffer
 	}
 
+	if publication == nil || !publication.IsConnected() {
```

Comment:
where is this line in Java?

Reply:
https://github.com/real-logic/aeron/blob/8dccf9dca51cbad3aa4f7a3abf17dbedc5a90199/aeron-cluster/src/main/java/io/aeron/cluster/service/ClusteredServiceAgent.java#L646-L649

```
        if (null == publication)
        {
            return Publication.NOT_CONNECTED;
        }
```

I think i just try to be a smart alec and be defensive here. I can remove if needed

Reply:
I see, LGTM